vivisect/symboliks/tests/test_analysis.py

```diff
@@ -38,3 +38,77 @@ def test_getStackOffset_below(self, addr=0xbfbff000, size=16384):
         offset = self.sfe.getStackOffset(MockVar(addr-size))
 
         self.assertIs(offset, None)
+
+
+import sys
+import vivisect.tests.vivbins as vivbins
+from vivisect.tests.vivbins import getTestWorkspace, getAnsWorkspace
+def cb_astNodeCount(path,obj,ctx):
+    ctx['count'] += 1
+    if len(path) > ctx['depth']:
+        ctx['depth'] = len(path)
+    print "\n\t%r\n\t\t%s" % (obj, '\n\t\t'.join([repr(x) for x in path]))
+
+
+class WalkTreeTest(unittest.TestCase):
+
+    @vivbins.require
+    def test_symbolik_maneuvers(self):
+        try:
+            vw = getTestWorkspace('test_kernel32_32bit-5.1.2600.5781.dll')
+            walkTreeDoer(vw)
+        except Exception as e:
+            sys.excepthook(*sys.exc_info())
+
+        try:
+            vw = getTestWorkspace('test_elf_i386')
+            walkTreeDoer(vw)
+        except Exception as e:
+            sys.excepthook(*sys.exc_info())
+
+
+        
+def walkTreeDoer(vw):
+    sctx = vsym_analysis.getSymbolikAnalysisContext(vw)
+    print sctx
+
+
+    count = 0
+    for fva in vw.getFunctions():
+        ctx = {'depth':0, 'count':0}
+        count += 1
+        print "(%d) 0x%x done" % (count, fva)
+        #raw_input("============================================================")
+
+        for spath in sctx.getSymbolikPaths(fva, maxpath=1):
+            effs = spath[-1]
+            if not len(effs):
+                continue
+            eff = effs[-1]
+
+            print "=====\n %r \n=====" % (eff)
+            # this is ugly
```

Comment:
actually, this is the state of the system at this point.  i can remove the unit test, but creating a unified api is well outside the scope of this PR ;)  i'd be happy to add this to my todo list if you like, or you can do it... but i wouldn't hold up this PR for this.  i agree that it's unmaintainable, but it should be fixed elsewhere.  this is the current state.